### PR TITLE
[blockly] add new list, item, persistence, text blocks

### PR DESCRIPTION
**Pull request by stefan-hoehn**
Signed-off-by: Stefan Höhn <stefan@andreaundstefanhoehn.de>

oh_text_replace: a new text replace block that is missing in the standard text block section
image

oh_taggeditems: retrieve a list of items identified by one or multiple tags
image

oh_list_concatenate: allows to concatenate two lists into a new lists
image

oh_get_persistvalue: add new previousState (new mutating block)

image

fixed type checking of oh_get_persistvalue: requires item name, not item

oh_zdt_fromText: Add time support 
image

oh_transformation: fix help url

see some more info and example here: https://community.openhab.org/t/extending-blockly-with-new-openhab-commands/127169/394

### CODE REVIEW

**Comment by stefan-hoehn** [comment]
@ghys Do you mind having a quick look at it, so we can deliver it with the current milestone

**Comment by stefan-hoehn** [comment]
@lolodomo would you be able to review it, so there is a chance that it still can make it into main before the feature freeze?

**Comment by ghys** [comment]
I'm still around... I'll review it and other pending PRs this week-end.

**Comment by stefan-hoehn** [comment]
> I'm still around... I'll review it and other pending PRs this week-end.

Yannick, I know, I appreciate it. please take care of yourself.! 🤗

**Comment by stefan-hoehn** [comment]
FYI I took the chance and added a few changes lately and a fix all of which are documented above. The reason is that by using the blocks myself I discovered some short comings, so I improved them.

The PR is now stable again and ready to be reviewed.

File: bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js [hunk 1]
```diff
@@ -55,7 +57,16 @@ export function addDateSupport () {
...
+      '  var now = zdt.now();',
+      '  var now_year = now.getYear();',
+      '  var now_month = now.getMonthValue();',
+      '  var now_day = now.getDayOfMonth();',
+      '  var today = \'\' + now_year;',
+      '  today += \'-\' + (\'0\' + now_month).slice(-2);',
+      '  today += \'-\' + (\'0\' + now_day).slice(-2)+\' \';',
       '  switch (true) {',
+      `    case regex_time_min.test(datetime): return ${zdt}.parse(today+datetime+':00+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
+      `    case regex_time_sec.test(datetime): return ${zdt}.parse(today+datetime+'+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js:69]:
```suggestion
      `    case regex_time_min.test(datetime): return ${zdt}.parse(today + datetime+':00+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
      `    case regex_time_sec.test(datetime): return ${zdt}.parse(today + datetime+'+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
```

**Review by ghys — CHANGES_REQUESTED** [review]
You're getting good at this, I have just this to moan about but otherwise it's all good.

**Comment by stefan-hoehn** [comment]
@ghys Your change has been applied.

**Review by ghys — APPROVED** [review]
There was still missing spaces that I didn't fix in the earlier suggestion, I did it myself.

LGTM thanks.

**Pull request merged**